### ISSUE TITLE
SC2034 unused non-existing variable

### ISSUE THREAD

**Issue by paranoicsan** [koalaman/shellcheck#1119]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034, SC1068
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

Shellcheck has been built on Codeship stack with Ubuntu inside

```sh

cabal update
cabal install shellcheck --reinstall --force-reinstall
[...]
Registering ShellCheck-0.4.7...
Installed ShellCheck-0.4.7

``` 

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
bundle exec rspec

```

#### Here's what shellcheck currently says:

```
In .codeship/tests.sh line 3:
bundle exec rspec
^-- SC2034: bundle appears unused. Verify it or export it.
       ^-- SC1068: Don't put spaces around the = in assignments.
```

#### Here's what I wanted or expected to see:

No issues should be found.

**Comment by Nightfirecat** [comment]
I'm unable to reproduce on my local system (also using shellcheck 0.4.7), or on the website.

**Comment by paranoicsan** [comment]
@Nightfirecat thanks for the response. Is there any additional information I can gather to try to narrow the possible reason?

**Comment by Nightfirecat** [comment]
I'm not sure... This definitely sounds like it could be an environment issue, but I'm not too familiar with how that might affect shellcheck's behavior myself.

**Comment by koalaman** [comment]
Wow, that's weird. It's complaining about line 3 of a two line file, and includes two comments: one that appears to match the line and one that does not.

1. What's the exact command you're using to invoke shellcheck? (Include the value of `$SHELLCHECK_OPTS` if set)
1. Does this happen if you copy-paste the code from this issue into a new file, and just run `shellcheck thatfile`?
1. Can you ensure it calls the Cabal binary directly, and not an alias/function/wrapper or previous version?
1. Is there any chance this file was modified while ShellCheck was running?

**Comment by paranoicsan** [comment]
Ok, here's what I gathered. 

There were 3 lines of code in the sample file, not 2 - my mistake.
```sh
#!/usr/bin/env bash

bundle exec rspec
```

> What's the exact command you're using to invoke shellcheck? (Include the value of $SHELLCHECK_OPTS if set)

`shellcheck` is building using [this script](https://raw.githubusercontent.com/codeship/scripts/master/packages/shellcheck.sh). So I logged in the debug session and tried the following commands to invoke `shellcheck`

```sh
${HOME}/cache/shellcheck .codeship/tests.sh
${HOME}/bin/shellcheck .codeship/tests.sh
```

No luck 😞 

> Does this happen if you copy-paste the code from this issue into a new file

Yes

> Can you ensure it calls the Cabal binary directly, and not an alias/function/wrapper or previous version?

Yes. The preinstalled version is `0.3.4` and it does not show such errors.

> Is there any chance this file was modified while ShellCheck was running?

No

> What's the OS and GHC version you're using?

`The Glorious Glasgow Haskell Compilation System, version 7.6.3`
`4.4.0-112-generic Ubuntu`

I could stick with `0.3.4` but it shows errors in another script:

```sh
#!/usr/bin/env bash

[...]

RAILS_ENV="test" bundle exec rake db:migrate

[...]

RAILS_ENV="test" bundle exec rake assets:precompile
```

Shows

```sh
In .codeship/setup.sh line 30:
RAILS_ENV="test" bundle exec rake db:migrate
^-- SC2037: To assign the output of a command, use var=$(cmd) .


In .codeship/setup.sh line 39:
RAILS_ENV="test" bundle exec rake assets:precompile
^-- SC2037: To assign the output of a command, use var=$(cmd) .
```

**Comment by koalaman** [comment]
Wow, I'm able to reproduce this on Ubuntu 14.04. This happens even with just `a b c`. The test suite fails dramatically.

I'm extremely reluctant to ever blame the compiler, but I see that this issue appears when compiled with `-O1` but not with `-O0` or in interpreted mode.

It works for me when I add `--enable-optimization=0` to your `cabal install` command.

Another approach is to fetch an officially compiled binary from https://shellcheck.storage.googleapis.com/index.html , which is built with a modern compiler. 7.6.3 is 5 years old.

**Comment by koalaman** [comment]
Closing since there's not a lot to be done on the shellcheck side.

**Comment by paranoicsan** [comment]
@koalaman thanks for the solution! I ended up fetching the latest binary from the resource you've provided 👍
